lib/spack/env/cc

```diff
@@ -339,7 +339,15 @@ case "$mode" in
         args=("${args[@]}" ${SPACK_LDLIBS[@]}) ;;
 esac
 
-full_command=("$command" "${args[@]}")
+#ccache only supports C languages, so filtering out Fortran
+if [[ ${lang_flags} != "F" && ${SPACK_CCACHE_BINARY} ]]; then
+    full_command=("${SPACK_CCACHE_BINARY}" "$command" "${args[@]}")
+    # #3761#issuecomment-294352232
+    # workaround for stage being a temp folder
+    export CCACHE_NOHASHDIR=yes
```

Comment:
Is there some way we could organize the temp space that would get around this issue?  The current stage directory structure could be simplified, and probably shouldn't track temp space *in* the Spack prefix.  We could move the symlinks to `~/.spack` and use a more deterministic naming scheme for the temp stages if that would help.